lib/actions/send-message.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const { sendMessage } = require('../sqs');
+const { toXml, toXmlError } = require('../utils/xml-helper');
+
+// TODO: Binary values in MessageAttributes have encoded value, AWS returns in encoded format when receiving message
+module.exports = body => {
+  const params = {
+    DelaySeconds: body.DelaySeconds,
+    MessageBody: body.MessageBody,
+    MessageGroupId: body.MessageGroupId,
+    QueueUrl: body.QueueUrl
+  };
+
+  const keys = Object.keys(body);
+
+  const messageAttributeKeys = keys.filter(key => key.startsWith('MessageAttribute'));
+
+  // how many exactly
+  const objs = messageAttributeKeys.filter(key =>
+    /MessageAttribute\.(\d{2}|\d{1})\.Name/.exec(key)
+  );
+
+  if (objs.length > 10) {
+    throw Object.assign(
+      new Error('Number of message attributes [12] exceeds the allowed maximum [10].'),
+      {
+        code: 'InvalidParameterValue'
+      }
+    );
+  }
+
+  const messageAttributes = {};
```

Comment:
When iterating and building a new object, consider **reduce**


  const messageAttributes = matchingObjects.reduce((acc, obj, index) => {
    const i = index + 1; // Starts at 1
    acc[body[`MessageAttribute.${i}.Name`]] = {
      DataType: body[`MessageAttribute.${i}.Value.DataType`]
    };

    if (body[`MessageAttribute.${i}.Value.StringValue`]) {
      acc[body[`MessageAttribute.${i}.Name`]].StringValue =
        body[`MessageAttribute.${i}.Value.StringValue`];
    }

    if (body[`MessageAttribute.${i}.Value.BinaryValue`]) {
      acc[body[`MessageAttribute.${i}.Name`]].BinaryValue =
        body[`MessageAttribute.${i}.Value.BinaryValue`];
    }

    return acc;
  }, {});